### PR TITLE
Issue #686 - Removes excludes of PMD

### PR DESCRIPTION
**Pull request by dalifreire**
Issue #686. After upgrade to qulice 0.16.4 we have to fix AvoidDuplicateLiterals PMD warings in several classes. This PR do the following changes:
- Removes PMD exclusions from pom.xml.
- Fix the PMD warnings in several classes that are removed from PMD exclusions.
- Three new puzzles were created.

### CODE REVIEW

**Comment by dalifreire** [comment]
@davvd Please, find someone to review.

**Comment by davvd** [comment]
@dalifreire Many thanks for the PR, let me find a reviewer for it

**Comment by davvd** [comment]
@pinaf this pull request is for you,please review

File: src/main/java/org/takes/facets/auth/social/PsLinkedin.java [hunk 1]
```diff
@@ -110,7 +115,7 @@ public PsLinkedin(final Href thref, final Href ahref,
         throws IOException {
         final Href href = new RqHref.Base(request).href();
         // @checkstyle MultipleStringLiteralsCheck (1 line)
```
Comment (pinaf) [inline src/main/java/org/takes/facets/auth/social/PsLinkedin.java]: @dalifreire we could remove this one also

File: src/main/java/org/takes/HttpException.java [hunk 2]
```diff
@@ -61,7 +66,7 @@ public HttpException(final int code) {
      */
     public HttpException(final int code, final String cause) {
         // @checkstyle MultipleStringLiteralsCheck (1 line)
```
Comment (pinaf) [inline src/main/java/org/takes/HttpException.java]: @dalifreire we could probably remove this suppression also

File: src/main/java/org/takes/facets/flash/RsFlash.java [hunk 3]
```diff
@@ -100,12 +100,21 @@
  * @author Yegor Bugayenko (yegor@teamed.io)
  * @version $Id$
  * @since 0.1
+ * @todo #686:30min After upgrade to qulice 0.16.4 we have to fix
+ *  SingularField PMD warings in this classe. Fix this warning
```
Comment (pinaf) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @dalifreire typo `class`

File: src/main/java/org/takes/facets/flash/RsFlash.java [hunk 4]
```diff
@@ -184,7 +193,7 @@ public RsFlash(final String msg, final Level level, final String cookie)
         throws UnsupportedEncodingException {
         super(RsFlash.make(msg, level, cookie));
         // @checkstyle MultipleStringLiteralsCheck (1 line)
```
Comment (pinaf) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @dalifreire we can probably remove this suppresion

File: src/main/java/org/takes/facets/flash/RsFlash.java [hunk 5]
```diff
@@ -100,12 +100,21 @@
  * @author Yegor Bugayenko (yegor@teamed.io)
  * @version $Id$
  * @since 0.1
+ * @todo #686:30min After upgrade to qulice 0.16.4 we have to fix
```
Comment (pinaf) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @dalifreire let's shorten this one to 15m

File: src/main/java/org/takes/rq/RqRequestLine.java [hunk 6]
```diff
@@ -194,7 +201,7 @@ private static Matcher matcher(final String line)
                     HttpURLConnection.HTTP_BAD_REQUEST,
                     String.format(
                         // @checkstyle MultipleStringLiteralsCheck (1 line)
```
Comment (pinaf) [inline src/main/java/org/takes/rq/RqRequestLine.java]: @dalifreire we can probably remove this suppresion

File: src/main/java/org/takes/rq/RqRequestLine.java [hunk 7]
```diff
@@ -216,7 +223,7 @@ private static String validated(final String line)
...
```
Comment (pinaf) [inline src/main/java/org/takes/rq/RqRequestLine.java]: @dalifreire we can probably remove this suppresion

File: src/main/java/org/takes/tk/TkWrap.java [hunk 8]
```diff
@@ -39,6 +39,7 @@
  * @version $Id$
  * @since 0.4
  */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
```
Comment (pinaf) [inline src/main/java/org/takes/tk/TkWrap.java]: @dalifreire if this one has duplicate literals problems, we should add it to the exclude list in `pom.xml` so the dev that gets the puzzle can work on it also.

File: src/test/java/org/takes/facets/auth/social/PsGoogleTest.java [hunk 9]
```diff
@@ -57,8 +57,48 @@
 public final class PsGoogleTest {
 
     /**
+     * Octocat.
+     */
+    private static final String OCTOCAT = "octocat";
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsGoogleTest.java]: @dalifreire This constant should be different in each test since it is not a fixed google API value. Please see http://www.yegor256.com/2016/05/03/test-methods-must-share-nothing.html for further info.

File: src/test/java/org/takes/facets/auth/social/PsGoogleTest.java [hunk 10]
```diff
@@ -57,8 +57,48 @@
...
+     */
+    private static final String URL = "url";
+    /**
+     * Act head.
+     */
+    private static final String ACT_HEAD = "GET /plus/v1/people/me";
+    /**
+     * Identity urn.
+     */
+    private static final String IDENTITY_URN = "urn:google:1";
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsGoogleTest.java]: @dalifreire this one can also probably be different for each test.

File: src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java [hunk 11]
```diff
@@ -54,10 +54,35 @@
...
+     */
+    private static final String FIRST_NAME = "firstName";
+    /**
+     * Last name.
+     */
+    private static final String LAST_NAME = "lastName";
+    /**
+     * Frodo.
+     */
+    private static final String FRODO = "Frodo";
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java [hunk 12]
```diff
@@ -54,10 +54,35 @@
...
+    /**
+     * Baggins.
+     */
+    private static final String BAGGINS = "Baggins";
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java [hunk 13]
```diff
@@ -54,10 +54,35 @@
...
+ *  ExcessiveMethodLength PMD warings in this classe. Fix this warning
+ *  and rid of the PMD suppression here.
  */
+@SuppressWarnings("PMD.ExcessiveMethodLength")
 public final class PsLinkedinTest {
 
     /**
+     * First name.
+     */
+    private static final String FIRST_NAME = "firstName";
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java [hunk 14]
```diff
@@ -54,10 +54,35 @@
...
+    /**
+     * Last name.
+     */
+    private static final String LAST_NAME = "lastName";
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/facets/auth/social/PsTwitterTest.java [hunk 15]
```diff
@@ -50,6 +50,15 @@
 public final class PsTwitterTest {
 
     /**
+     * Http ok.
+     */
+    private static final String HTTP_OK = "HTTP OK";
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsTwitterTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/facets/auth/social/PsTwitterTest.java [hunk 16]
```diff
@@ -50,6 +50,15 @@
...
+    /**
+     * Name.
+     */
+    private static final String NAME = "name";
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsTwitterTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/facets/fork/TkConsumesTest.java [hunk 17]
```diff
@@ -50,20 +50,29 @@
 public final class TkConsumesTest {
 
     /**
+     * Application json.
+     */
+    private static final String APPLICATION_JSON = "application/json";
+    /**
+     * Content-type json.
+     */
+    private static final String CT_JSON = "Content-Type: application/json";
```
Comment (pinaf) [inline src/test/java/org/takes/facets/fork/TkConsumesTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/http/BkBasicTest.java [hunk 18]
```diff
@@ -284,7 +290,7 @@ public void run() {
                         BkBasicTest.POST,
                         BkBasicTest.HOST,
                         "",
-                        "Hello World!"
+                        BkBasicTest.HELLO_WORLD
```
Comment (pinaf) [inline src/test/java/org/takes/http/BkBasicTest.java]: @dalifreire It looks like we can change this string to something else that is specific to this test

File: src/test/java/org/takes/http/BkBasicTest.java [hunk 19]
```diff
@@ -343,7 +349,7 @@ public void run() {
                         BkBasicTest.HOST,
                         "Connection: Close",
                         "",
-                        "Hello World!"
+                        BkBasicTest.HELLO_WORLD
```
Comment (pinaf) [inline src/test/java/org/takes/http/BkBasicTest.java]: @dalifreire It looks like we can change this string to something else that is specific to this test

File: src/test/java/org/takes/rq/RqWithHeadersTest.java [hunk 20]
```diff
@@ -40,6 +40,14 @@
  */
 public final class RqWithHeadersTest {
     /**
+     * Test header.
+     */
+    private static final String TEST_HEADER = "TestHeader: someValue";
```
Comment (pinaf) [inline src/test/java/org/takes/rq/RqWithHeadersTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/rq/RqWithHeadersTest.java [hunk 21]
```diff
@@ -40,6 +40,14 @@
...
+    /**
+     * Some header.
+     */
+    private static final String SOME_HEADER = "SomeHeader: testValue";
```
Comment (pinaf) [inline src/test/java/org/takes/rq/RqWithHeadersTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/rq/multipart/RqMtBaseTest.java [hunk 22]
```diff
@@ -46,6 +46,35 @@
...
+    private static final String BODY_ELEMENT = "--AaB01x";
+    /**
+     * Content type.
+     */
+    private static final String CONTENT_TYPE =
+        "Content-Type: multipart/form-data; boundary=AaB01x";
+    /**
+     * Host example.
+     */
+    private static final String HOST_EXAMPLE = "Host: rtw.example.com";
```
Comment (pinaf) [inline src/test/java/org/takes/rq/multipart/RqMtBaseTest.java]: @dalifreire this can probably be a local test variable, different for each test

File: src/test/java/org/takes/rq/multipart/RqMtBaseTest.java [hunk 23]
```diff
@@ -46,6 +46,35 @@
...
+     */
+    private static final String FORM_DATA = "form-data; name=\"%s\"";
+    /**
+     * POST head.
+     */
+    private static final String POST_HEAD = "POST /h?a=4 HTTP/1.1";
+    /**
+     * Content length.
+     */
+    private static final String LENGTH = "Content-Length: 100007";
```
Comment (pinaf) [inline src/test/java/org/takes/rq/multipart/RqMtBaseTest.java]: @dalifreire this can probably be a local test variable, different for each test

File: src/test/java/org/takes/rq/multipart/RqMtBaseTest.java [hunk 24]
```diff
@@ -46,6 +46,35 @@
...
+     */
+    private static final String HOST_EXAMPLE = "Host: rtw.example.com";
+    /**
+     * Form data.
+     */
+    private static final String FORM_DATA = "form-data; name=\"%s\"";
+    /**
+     * POST head.
+     */
+    private static final String POST_HEAD = "POST /h?a=4 HTTP/1.1";
```
Comment (pinaf) [inline src/test/java/org/takes/rq/multipart/RqMtBaseTest.java]: @dalifreire this can probably be a local test variable, different for each test

File: src/test/java/org/takes/rq/multipart/RqMtBaseTest.java [hunk 25]
```diff
@@ -46,6 +46,35 @@
...
+    /**
+     * Content length.
+     */
+    private static final String LENGTH = "Content-Length: 100007";
+    /**
+     * Address.
+     */
+    private static final String ADDRESS = "447 N Wolfe Rd, Sunnyvale, CA 94085";
```
Comment (pinaf) [inline src/test/java/org/takes/rq/multipart/RqMtBaseTest.java]: @dalifreire this can probably be a local test variable, different for each test

File: src/test/java/org/takes/rq/multipart/RqMtFakeTest.java [hunk 26]
```diff
@@ -46,6 +46,36 @@
  */
 public final class RqMtFakeTest {
     /**
+     * Bar.
+     */
+    private static final String BAR = "bar";
```
Comment (pinaf) [inline src/test/java/org/takes/rq/multipart/RqMtFakeTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/rq/multipart/RqMtFakeTest.java [hunk 27]
```diff
@@ -46,6 +46,36 @@
...
+    /**
+     * Content.
+     */
+    private static final String CONTENT = "content";
```
Comment (pinaf) [inline src/test/java/org/takes/rq/multipart/RqMtFakeTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/rq/multipart/RqMtFakeTest.java [hunk 28]
```diff
@@ -46,6 +46,36 @@
...
+    /**
+     * Closed.
+     */
+    private static final String CLOSED = "Closed";
```
Comment (pinaf) [inline src/test/java/org/takes/rq/multipart/RqMtFakeTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/rq/multipart/RqMtFakeTest.java [hunk 29]
```diff
@@ -46,6 +46,36 @@
...
+    /**
+     * Name.
+     */
+    private static final String NAME = "name";
```
Comment (pinaf) [inline src/test/java/org/takes/rq/multipart/RqMtFakeTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/rq/multipart/RqMtFakeTest.java [hunk 30]
```diff
@@ -46,6 +46,36 @@
...
+    /**
+     * Form data.
+     */
+    private static final String FORM_DATA =
+        "form-data; name=\"data\"; filename=\"%s\"";
+    /**
+     * IOException message.
+     */
+    private static final String IOEXCEPTION_MSG =
```
Comment (pinaf) [inline src/test/java/org/takes/rq/multipart/RqMtFakeTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/rq/multipart/RqMtFakeTest.java [hunk 31]
```diff
@@ -46,6 +46,36 @@
...
+        "An IOException was expected since the Stream is closed";
+    /**
+     * Foo.
+     */
+    private static final String FOO = "foo";
```
Comment (pinaf) [inline src/test/java/org/takes/rq/multipart/RqMtFakeTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/rq/multipart/RqMtSmartTest.java [hunk 32]
```diff
@@ -62,6 +62,19 @@
  */
 public final class RqMtSmartTest {
     /**
+     * Post.
+     */
+    private static final String POST = "POST /post?u=3 HTTP/1.1";
```
Comment (pinaf) [inline src/test/java/org/takes/rq/multipart/RqMtSmartTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 33]
```diff
@@ -38,6 +38,35 @@
...
+     */
+    private static final String WORKS = "works?";
+    /**
+     * Set cookie.
+     */
+    private static final String SET_COOKIE = "Set-Cookie: foo=works?;Path=/;";
+    /**
+     * Foo.
+     */
+    private static final String FOO = "foo";
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 34]
```diff
@@ -38,6 +38,35 @@
 public final class RsWithCookieTest {
 
     /**
+     * Carriage return constant.
+     */
+    private static final String CRLF = "\r\n";
+    /**
+     * Bar.
+     */
+    private static final String BAR = "bar";
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 35]
```diff
@@ -38,6 +38,35 @@
...
+    /**
+     * Http ok.
+     */
+    private static final String HTTP_OK = "HTTP/1.1 200 OK";
+    /**
+     * Works.
+     */
+    private static final String WORKS = "works?";
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire this can probably be a local test variable

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 36]
```diff
@@ -48,13 +77,15 @@ public void addsCookieToResponse() throws IOException {
...
+                    RsWithCookieTest.FOO,
+                    RsWithCookieTest.WORKS,
+                    RsWithCookieTest.PATH
                 )
             ).print(),
             Matchers.equalTo(
-                Joiner.on("\r\n").join(
-                    "HTTP/1.1 200 OK",
-                    "Set-Cookie: foo=works?;Path=/;",
+                Joiner.on(RsWithCookieTest.CRLF).join(
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire let's move this joiner logic into a private method

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 37]
```diff
@@ -38,6 +38,35 @@
 public final class RsWithCookieTest {
 
     /**
+     * Carriage return constant.
+     */
+    private static final String CRLF = "\r\n";
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire we could get rid of this variable

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 38]
```diff
@@ -38,6 +38,35 @@
...
+    /**
+     * Bar.
+     */
+    private static final String BAR = "bar";
+    /**
+     * Http ok.
+     */
+    private static final String HTTP_OK = "HTTP/1.1 200 OK";
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire and get rid of this one too

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 39]
```diff
@@ -38,6 +38,35 @@
...
+    /**
+     * Works.
+     */
+    private static final String WORKS = "works?";
+    /**
+     * Set cookie.
+     */
+    private static final String SET_COOKIE = "Set-Cookie: foo=works?;Path=/;";
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire and this one should be a string literal inside the new private method. we can make the `works?` part a method parameter.

File: src/main/java/org/takes/facets/auth/social/PsLinkedin.java (hunk 1 above)
Comment (dalifreire) [inline src/main/java/org/takes/facets/auth/social/PsLinkedin.java]: @pinaf Fixed!

File: src/main/java/org/takes/HttpException.java (hunk 2 above)
Comment (dalifreire) [inline src/main/java/org/takes/HttpException.java]: @pinaf Fixed!

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 3 above)
Comment (dalifreire) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @pinaf Fixed!

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 4 above)
Comment (dalifreire) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @pinaf Fixed!

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 5 above)
Comment (dalifreire) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @pinaf Fixed!

File: src/main/java/org/takes/rq/RqRequestLine.java (hunk 6 above)
Comment (dalifreire) [inline src/main/java/org/takes/rq/RqRequestLine.java]: @pinaf Fixed!

File: src/main/java/org/takes/rq/RqRequestLine.java (hunk 7 above)
Comment (dalifreire) [inline src/main/java/org/takes/rq/RqRequestLine.java]: @pinaf Fixed!

File: src/main/java/org/takes/tk/TkWrap.java (hunk 8 above)
Comment (dalifreire) [inline src/main/java/org/takes/tk/TkWrap.java]: @pinaf Fixed!

File: src/test/java/org/takes/facets/auth/social/PsGoogleTest.java [hunk 40]
```diff
@@ -113,13 +147,13 @@ public Response act(final Request req) throws IOException {
                         );
                         return new RsJson(
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsGoogleTest.java]: @dalifreire if we extract the creation of this `RsJson` into a private method, we could get rid of `IMAGE` and `URL`, reducing coupling between tests. This method should expect the `id` as a parameter.

File: src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java [hunk 41]
```diff
@@ -111,9 +120,13 @@ public Response act(final Request req) throws IOException {
                         return new RsJson(
                             Json.createObjectBuilder()
                                 .add("id", identifier)
-                                .add("firstName", "Frodo")
-                                .add("lastName", "Baggins")
-                                .build()
+                                .add(
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java]: @dalifreire let's make this into a single line `.add(firstname, frodo)`

File: src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java [hunk 42]
```diff
@@ -111,9 +120,13 @@ public Response act(final Request req) throws IOException {
...
+                                    firstname,
+                                    frodo
+                                ).add(
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java]: @dalifreire same here

File: src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java [hunk 43]
```diff
@@ -146,8 +159,14 @@ public void exec(final URI home) throws IOException {
                     MatcherAssert.assertThat(
                         identity.properties(),
                         Matchers.allOf(
-                            Matchers.hasEntry("firstName", "Frodo"),
-                            Matchers.hasEntry("lastName", "Baggins")
+                            Matchers.hasEntry(
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java]: @dalifreire single line

File: src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java [hunk 44]
```diff
@@ -146,8 +159,14 @@ public void exec(final URI home) throws IOException {
...
+                                firstname,
+                                frodo
+                            ),
+                            Matchers.hasEntry(
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java]: @dalifreire single line

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 45]
```diff
@@ -69,20 +87,26 @@ public void addsCookieToResponse() throws IOException {
...
+        final String value = "value?";
         MatcherAssert.assertThat(
             new RsPrint(
                 new RsWithCookie(
                     new RsWithCookie(
                         new RsEmpty(),
-                        "foo", "works?", "Path=/"
+                        xoo,
+                        value,
+                        RsWithCookieTest.PATH
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire I think we could have another unique path here and get rid of the shared `PATH` variable

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 46]
```diff
@@ -104,6 +128,7 @@ public void rejectsInvalidName() {
      */
     @Test(expected = IllegalArgumentException.class)
     public void rejectsInvalidValue() {
-        new RsWithCookie(new RsEmpty(), "bar", "wo\"rks");
+        final String cookiename = "cookiename";
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire this variable is unnecessary, we can inline it

**Comment by dalifreire** [comment]
@pinaf Thanks for your review. Could you check if is better now?

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 47]
```diff
@@ -104,6 +106,68 @@ public void rejectsInvalidName() {
...
+    }
+
+    /**
+     * Returns the joined cookie.
+     * @param name Cookie name
+     * @param value Cookie value
+     * @param path Cookie path
+     * @return Joined cookie
+     */
+    private String joiner(final String name, final String value,
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire can't we have just 1 joiner method that takes multiple cookie strings ?

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 48]
```diff
@@ -104,6 +106,68 @@ public void rejectsInvalidName() {
...
+        final String path) {
+        final String[] parts = {
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire we can inline these arrays, no?

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 49]
```diff
@@ -104,6 +106,68 @@ public void rejectsInvalidName() {
...
+    private String cookie(final String name, final String value,
+        final String path) {
+        return String.format("Set-Cookie: %s=%s;%s;", name, value, path);
+    }
+    /**
+     * Returns the http ok.
+     * @return HTTP OK
+     */
+    private String httpOk() {
+        return "HTTP/1.1 200 OK";
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire if we have a single joiner method, this could go away

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 50]
```diff
@@ -104,6 +106,68 @@ public void rejectsInvalidName() {
...
+    }
+    /**
+     * Returns the carriage return joiner.
+     * @return Joiner
+     */
+    private Joiner joiner() {
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire and this one also

**Comment by dalifreire** [comment]
@pinaf Thanks again. Please, could you check if everything is right now?

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 51]
```diff
@@ -104,6 +107,34 @@ public void rejectsInvalidName() {
...
-        new RsWithCookie(new RsEmpty(), "bar", "wo\"rks");
+        new RsWithCookie(new RsEmpty(), "cookiename", "wo\"rks");
+    }
+
+    /**
+     * Returns the joined cookie.
+     * @param cookies Cookies values
+     * @return Joined cookie
+     */
+    private static String joiner(final String... cookies) {
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire let's rename this to `cookies`

File: src/test/java/org/takes/rs/RsWithCookieTest.java [hunk 52]
```diff
@@ -69,23 +69,26 @@ public void addsCookieToResponse() throws IOException {
...
             ).print(),
             Matchers.equalTo(
-                Joiner.on("\r\n").join(
-                    "HTTP/1.1 200 OK",
-                    "Set-Cookie: foo=works?;Path=/;",
-                    "Set-Cookie: bar=worksToo?;Path=/2nd/path/;",
-                    "",
-                    ""
+                joiner(
+                    formatCookie(qux, value, path),
```
Comment (pinaf) [inline src/test/java/org/takes/rs/RsWithCookieTest.java]: @dalifreire let's call this using the full path `RsWithCookieTest.formatCookie` and do the same for the `joiner` method calls.

File: src/test/java/org/takes/facets/auth/social/PsGoogleTest.java [hunk 53]
```diff
@@ -112,18 +138,7 @@ public Response act(final Request req) throws IOException {
...
-                                .add(
-                                    "image",
-                                    Json.createObjectBuilder()
-                                        .add(
-                                            "url",
-                                            PsGoogleTest.AVATAR
-                                        )
-                                )
-                                .build()
+                            jsonBuilder("1").add("displayName", octocat).build()
```
Comment (pinaf) [inline src/test/java/org/takes/facets/auth/social/PsGoogleTest.java]: @dalifreire sorry, this suggestion was a mistake. could you revert the changes done to this class in commit https://github.com/yegor256/takes/pull/689/commits/b272c20249759b8939503b4ed70b4bc17b36a87e ?

**Comment by pinaf** [comment]
@dalifreire just 3 more and we are done

**Comment by pinaf** [comment]
@rultor merge

**Comment by dalifreire** [comment]
@yegor256 ping...

**Comment by yegor256** [comment]
@rultor merge

**Comment by dalifreire** [comment]
@davvd Please, proceed

**Comment by davvd** [comment]
@elenavolokhova please, check this issue for QA compliance, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor deploy pls

**Comment by elenavolokhova** [comment]
@davvd  Looks good!

**Comment by davvd** [comment]
> @davvd  Looks good!

@elenavolokhova thanks, got it, great to see high-quality tickets here

**Comment by davvd** [comment]
@pinaf just added 10 mins to @elenavolokhova (for QA), payment ID is `91825740`

Many thanks! **1 hour and 10 mins** were added to your account in Transaction ID `AP-25A13869HX7259925` (task took 265 hours and 24 mins)

review comments (c=55) added as a bonus

+70 added to your rating, current score is: [+791](http://www.netbout.com/b/35128?open=rating)
